client/js/libs/handlebars/ircmessageparser/parseStyle.js

```diff
@@ -64,20 +94,23 @@ function parseStyle(text) {
 		case COLOR:
 			emitFragment();
 
+			// Go one step further to find the corresponding color
 			colorCodes = text.slice(position + 1).match(colorRx);
 
 			if (colorCodes) {
 				textColor = Number(colorCodes[1]);
-				bgColor = Number(colorCodes[2]);
-				if (Number.isNaN(bgColor)) {
-					bgColor = undefined;
+				if (colorCodes[2]) {
+					bgColor = Number(colorCodes[2]);
 				}
```

Comment:
Minor simplification, `colorCodes[2]` is either a number (by definition of the Regex) or `undefined`.